lib/senkyoshi/models/scorm_package.rb

```diff
@@ -1,52 +1,65 @@
 module Senkyoshi
   class ScormPackage
-    attr_accessor(:entries, :manifest)
+    attr_accessor(:entries, :manifest, :points_possible)
 
-    ##
-    # Scorm packages should include this string in the <schema> tag. We
-    # downcase, and remove spaces before checking to see if a manifest contains
-    # this schema to determine whether or not it belongs to a scorm package
-    ##
-    SCORM_SCHEMA = "adlscorm".freeze
-
-    def initialize(zip_file, manifest)
+    def initialize(zip_file, manifest, scorm_item = nil)
       @manifest = manifest
       @entries = ScormPackage.get_entries zip_file, manifest
+      @points_possible = if scorm_item
+                           scorm_item.xpath(
+                             "/scormItem/gradebookInfo/@pointsPossible",
+                           ).text
+                         end
     end
 
     ##
-    # Returns true if a manifest is a scorm manifest file, false otherwise
-    ##
-    def self.scorm_manifest?(manifest)
-      parsed_manifest = Nokogiri::XML(manifest.get_input_stream.read)
-      schema_name = parsed_manifest.
-        xpath("//xmlns:metadata/xmlns:schema").
-        text.delete(" ").downcase
-      return schema_name == SCORM_SCHEMA
-    # NOTE we occasionally run into malformed manifest files
-    rescue Nokogiri::XML::XPath::SyntaxError => e
-      filename = manifest.zipfile
-      STDERR.puts "Malformed scorm manifest found: #{manifest} in #{filename}"
-      STDERR.puts e.to_s
-      false
+    # Extracts scorm packages from a blackboard export zip file
+    ##
+    def self.get_scorm_packages(blackboard_export)
+      find_scorm_items(blackboard_export).
+        map do |item|
+          manifest_entry = find_scorm_manifest(blackboard_export, item)
+          ScormPackage.new blackboard_export, manifest_entry, item
+        end
     end
 
     ##
-    # Extracts scorm packages from a blackboard export zip file
+    # Returns paths to scormItem files
     ##
-    def self.get_scorm_packages(blackboard_export)
-      find_scorm_manifests(blackboard_export).map do |manifest|
-        ScormPackage.new blackboard_export, manifest
+    def self.find_scorm_item_paths(zip_file)
+      Nokogiri::XML.parse(
+        Senkyoshi.read_file(zip_file, "imsmanifest.xml"),
+      ).
+        xpath("//resource[@type='resource/x-plugin-scormengine']").
+        map { |r| r.xpath("./@bb:file").text }
+    rescue Exceptions::MissingFileError
+      []
```

Comment:
You probably should be logging to STDERR instead of just swallowing the exception.